Signon support improvements

cbaines opened the pull request:
These changes improve the support for running the Publishing E2E Tests where the applications under test use Signon for authentication. This is a continuation of the initial changes back in November [1], which implemented initial support for Signon aimed at allowing the tests to run (but not necessarily pass yet) when run in the Rails production environment.

1: https://github.com/alphagov/publishing-e2e-tests/pull/56

When running the tests with docker compose, this shouldn't have any effect, as Signon is not used. When running the tests with govuk-guix, not all of the pass, I get 11 passes, and 12 failures. The 5 Manuals Publisher tests all fail, I think due to issues with the test users Signon organisation. 6 tests fail between Specialist Publisher and Travel Advice Publisher due to 302 redirects, probably from authenticating proxy. There is also one test failure in the Publisher tests, as it doesn't expect the service to be running on a odd port.

kevindew commented on spec/support/collections_publisher_helpers.rb:
```diff
@@ -10,4 +10,18 @@ def fill_in_topic_form(slug:, title:, parent: nil)
   end
+
+  def self.included(base)
+    if SignonHelpers::use_signon?
```
might be slightly cleaner to have a guard clause and do `return unless SignonHelpers::use_signon?`

cbaines replied:
That works for me, I've updated all the support files.

cbaines commented:
> Changes look good to me. I notice though that it introduces single quotes strings into files that are using double quotes, would you be able to sort those?

Sure, I think I've got all the cases now. I've also changed the Travel Advice Publisher and Specialist Publisher support files to use guard clauses and double quotes.

> The thing that makes me a little anxious is that we're not actually running any of this code in the tests, so it might just slip out of date. Do you know what prerequisites there are to make use of this in tests?

Well, I run this code (as I switched govuk-guix across to running the Rails apps in the production environment back in November), but yes, at the time the only apps under test were Specialist Publisher and Travel Advice Publisher, so I only was able to add support to these apps. Obviously, things have slipped out of date a bit, hence this Pull Request.

As for prerequisites to use Signon in the tests, currently no tests depend on the availability of Signon, and adding Signon in to the mix actually causes a few of the existing tests to fail.

Back when I was writing the initial implementation, I was trying to think about the new possibilities that using Signon would bring, and I think there are a few cases where it could be really useful, access limitation may be one, as that works across many services, and involves Signon. I think the only prerequisites in term of more fully using support for Signon in the tests are getting all the existing tests passing when using Signon, and then maybe writing some tests that actually use some of the functionality that using Signon would offer.

If you're asking more about the environment side of this. For govuk-guix, you'll just need a more up-to-date copy than what is currently available publicly in the gds-attic. For using docker-compose with the configuration in this repository, I'm unsure, I did talk a little about this to Adrian and Ryan, so they may have more knowledge than I.

kevindew commented:
@cbaines Cool - well I'm happy for this to go in. I'll see if we can find some time in the e2e project to make use of it. I agree it'd be good to actually have signon used as part of the test, just not entirely sure it'll make it into the scope of the project and then what will happen after. 

Thanks for putting this together 👍

cbaines commented:
> Thanks for putting this together 👍

Great, thanks for reviewing :)
